### PR TITLE
Fix context load for non-docker endpoints

### PR DESCRIPTION
**Pull request by aiordache**
Skip non-docker endpoint parsing and add fallbacks for docker endpoint fields.

closes https://github.com/docker/docker-py/issues/2601

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (thaJeztah) [inline .vscode/launch.json:4]: Don't think these should have been included?

File: docker/api/build.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ def build(self, path=None, tag=None, quiet=False, fileobj=None,
                 with open(dockerignore, 'r') as f:
                     exclude = list(filter(
                         lambda x: x != '' and x[0] != '#',
-                        [l.strip() for l in f.read().splitlines()]
+                        [line.strip() for line in f.read().splitlines()]
```
Comment (thaJeztah) [inline docker/api/build.py:156]: unrelated change?

File: docker/context/context.py [hunk 3]
```diff
@@ -16,30 +16,42 @@ def __init__(self, name, orchestrator=None, host=None, endpoints=None,
...
+        # check docker endpoints
+        for k, v in endpoints.items():
+            if not isinstance(v, dict):
+                # unknown format
+                raise ContextException("""Unknown endpoint format for
+                    context {}: {}""".format(name, v))
+
+            self.endpoints[k] = v
+            if k != "docker":
+                continue
```
Comment (thaJeztah) [inline docker/context/context.py:49]: I wonder if `docker-py` should ignore these, or just load them. I'm not too familiar with the code, buta context has _some_ (semi-)strict fields, and other metadata that could be "anything". I think it should be ok to _load_ these, as long as they're not used to try to _connect_ with them.

(basically; if this library is just to _parse_ / _load_ contexts, it should not be too opinionated)

 WDYT?

File: docker/context/context.py [hunk 4]
```diff
@@ -16,30 +16,42 @@ def __init__(self, name, orchestrator=None, host=None, endpoints=None,
...
+
+            self.endpoints[k]["Host"] = v.get("Host", get_context_host(
+                host, tls))
+            self.endpoints[k]["SkipTLSVerify"] = bool(v.get(
+                "SkipTLSVerify", not tls))
```
Comment (thaJeztah) [inline docker/context/context.py:54]: I think this was the part where the problem occurred; would it be possible to ignore missing fields? (and use the default for those)? So if `SkipTLSVerify` is not present, then keep the default (`false`)

File: scripts/versions.py [hunk 5]
```diff
@@ -72,5 +72,6 @@ def main():
         results.add(str(latest))
     print(' '.join(results))
 
+
```
Comment (thaJeztah) [inline scripts/versions.py:75]: unrelated change

File: .vscode/launch.json (hunk 1 above)
Comment (aiordache) [reply .vscode/launch.json]: Yes, my mistake. Made the commit through vscode and didn't notice these files were also staged. Thanks!

File: docker/context/context.py (hunk 3 above)
Comment (aiordache) [reply docker/context/context.py:49]: They are loaded,  for docker endpoints we only verify that we have a Host and SkipTLSVerify fields set (otherwise we set some defaults).  For cloud endpoints we don't check anything.

**Comment by thaJeztah** [comment]
(oh, and pls squash the commits at the end 😅)

File: docker/context/context.py (hunk 4 above)
Comment (aiordache) [reply docker/context/context.py:54]: We do this with the **v.get()**. The 2nd parameter is the fallback value if the key is no found. The SkipTLSVerify fallback value depends on the specified tls certs(it should be false if we set certs otherwise True). For the host we need to detect the platform and set the path to the docker engine according to it.

Comment (thaJeztah) [reply docker/context/context.py:54]: > We do this with the v.get(). The 2nd parameter is the fallback value if the key is no found. 

Oh! See yes, the old code used `bool(v["SkipTLSVerify"])` and that caused the error

File: docker/context/context.py (hunk 3 above)
Comment (thaJeztah) [reply docker/context/context.py:49]: 🤦 I see it now, yes. I read the `continue` as "skip this endpoint", but it's loaded the line above, and only the special handling is skipped
